Redo cylinder

Gotr-byte opened the pull request.

pandaero reviewed (changes requested):
Makefile
- revert changes, don't remove those flags

minirt.h
- line 58: don't use multi-line comment when standard is single-line everywhere 🤦🏼 
- line 716: is this function in use?

intersect_cylinder.c
- remove comment line 16 (this can be there just for testing during testing)
- the defines lines 18-20 should be moved to minirt.h
- line 40: "solve" quadtatic equation
there is no way this is norminette-proof

test/
- these files should remain standard once the scene is meaningfully constructed (changing them is okay for testing **during** testing)

Gotr-byte commented:
minirt.h

line 58: don't use multi-line comment when standard is single-line everywhere 🤦🏼
-fixed

line 716: is this function in use?
-fixed / removed
intersect_cylinder.c

remove comment line 16 (this can be there just for testing during testing)
-fixed/removed
the defines lines 18-20 should be moved to minirt.h
-fixed/moved
line 40: "solve" quadtatic equation
there is no way this is norminette-proof
-ran norminette - did not show errors
test/
- added more tests
- Changes are seen in verbose commits, this version has 120 bytes of leaks when using the cylinder - i sent you the report.
- Intersections are on a norminette stage / final testing.
these files should remain standard once the scene is meaningfully constructed (changing them is okay for testing during testing)

pandaero commented:
Great, some functions are missing descriptions still, and file naming is inconsistent with the rest of the repo
